File: Front/lib/vote/vote.dart

class Vote {
  final int id;
  final int userId;
  final int taskId;
  final int value;

Vote({
    required this.id,
    required this.userId,
    required this.taskId,
    required this.value,
  });

  factory Vote.fromJson(Map<String, dynamic> json) {
    return Vote(
      id: json['ID'],
      userId: json['UserID'],
      taskId: json['TaskID'],
      value: json['Value'],
    );
  }
}

File: Back/model/vote.go
package model

import "gorm.io/gorm"

type Vote struct {
	gorm.Model
	ID     uint `gorm:"primaryKey"`
	UserID uint `gorm:"not null"`
	TaskID uint `gorm:"not null"`
	Value  int  `gorm:"not null"`
	User   User `gorm:"foreignKey:UserID" json:"-"`
	Task   Task `gorm:"foreignKey:TaskID" json:"-"`
}
